src/Rule/LatteTemplatesRule.php

```diff
@@ -159,24 +154,24 @@ private function analyseTemplates(array $templates, Scope $scope, array &$errors
 
             $dir = dirname($templatePath);
 
-            $collectedDataList = $fileAnalyserResult->getCollectedData();
             $includeTemplates = [];
-            foreach ($collectedDataList as $collectedData) {
-                if ($collectedData->getCollectorType() !== IncludePathCollector::class) {
-                    continue;
-                }
-                /** @phpstan-var CollectedIncludePathArray[] $dataList */
-                $dataList = $collectedData->getData();
-                foreach ($dataList as $data) {
-                    $collectedIncludedPath = CollectedIncludePath::fromArray($data, $this->typeStringResolver);
+            $collectedDataList = $fileAnalyserResult->getCollectedData();
+            foreach ($this->templateRenderCollector->extractCollectedData($fileAnalyserResult->getCollectedData()) as $collectedTemplateRender) {
```

Comment:
extraction of collected data moved to relevant collector to keep it together and hale Rule class cleaner